monkey/infection_monkey/telemetry/messengers/exploit_intercepting_telemetry_messenger.py

```diff
@@ -0,0 +1,19 @@
+from common.common_consts.telem_categories import TelemCategoryEnum
+from infection_monkey.telemetry.i_telem import ITelem
+from infection_monkey.telemetry.messengers.i_telemetry_messenger import ITelemetryMessenger
+from infection_monkey.tunnel import MonkeyTunnel
+
+
+class ExploitInterceptingTelemetryMessenger(ITelemetryMessenger):
+    def __init__(self, telemetry_messenger: ITelemetryMessenger, tunnel: MonkeyTunnel):
+        self._telemetry_messenger = telemetry_messenger
+        self._tunnel = tunnel
+
+    def send_telemetry(self, telemetry: ITelem):
+        if (
+            telemetry.telem_category == TelemCategoryEnum.EXPLOIT
+            and telemetry.propagation_result is True
```

Comment:
There's no `propagation_result` in `ITelem`

Reply:
Yeah, but we've already verified that it's an ExploitTelem. How do we work around this so the type checker is happy. Can we "cast" it somehow?

Reply:
Maybe I can avoid the issue with singledispatch.

Reply:
I ended up using singledispatch.